### ISSUE TITLE
Bump minor version when not needed

### ISSUE THREAD

**Issue by tleunen** [semantic-release/semantic-release#256]
I'm seeing the issue for a while now on a specific project, here: [tleunen/eslint-import-resolver-babel-module-alias](https://github.com/tleunen/eslint-import-resolver-babel-module-alias)

Any reason why it does that?

```
> eslint-import-resolver-babel-module-alias@0.0.0-semantically-released semantic-release /home/ubuntu/eslint-import-resolver-babel-module-alias
> semantic-release pre && npm publish && semantic-release post

+ eslint-import-resolver-babel-module-alias@1.4.0
Generating changelog from v1.3.0 to HEAD...
Parsed 0 commits.
```

**Comment by motiz88** [comment]
Seeing the same behavior here: [motiz88/astring-flow](https://github.com/motiz88/astring-flow). I'm also using CircleCI, maybe that's a factor? I have several other modules building on Travis with semantic-release, and that has always worked fine.

**Comment by motiz88** [comment]
**tl;dr: try running the attached `git`/`bash` one-liner prior to `npm publish`**

Got it! `semantic-release` is not the culprit (though it could possibly help - see 1 below). This happens when the "latest" version on npm is missing the `gitHead` field of `package.json`. In this case, `semantic-release` analyzes the _entire commit history_ each time, generating spurious version bumps.

This arises from in `npm publish`'s ~~brittle~~ _broken_ way of getting at that Git hash in the first place (PR [here](https://github.com/npm/read-package-json/pull/67)): It silently fails on perfectly valid Git repos if they use [packed refs](https://git-scm.com/docs/git-pack-refs) a certain way.

Curiously, _that_ somehow happens on CircleCI and not on, say, Travis (last time I checked). Haven't looked into why that is, since again, the situation is perfectly valid from a `git` standpoint - It's `npm` that's depending on an internal detail and thus breaking.

Now, the workaround: I added the following Bash command to `circle.yml` right before `npm run semantic-release`. It reconstructs the "unpacked refs" for all the branch heads that exist only as packed refs, so `npm` can read them where it expects them. Thus `npm publish` will include the correct `gitHead` reference and thus the _next_ `semantic-release` run will see the correct, truncated change list.

```bash
git show-ref --head --heads | while IFS=' ' read -r hash name; do test ! -e "${GIT_DIR:-.git}/$name" && echo $hash > "${GIT_DIR:-.git}/$name"; done
```

cc @peteyy @krishna-koushik re: #258 - maybe this applies to your cases too?

Other thoughts:

1. Two things `semantic-release` can (should?) do about this mess: emit a warning in case of a missing `gitHead`; infer the correct hash from the local repo by looking for the `vx.y.z` git tag auto-created by `npm publish`.

2. @bahmutov, do you feel like including the workaround in `condition-circle`? Seems like an easy enough fix. `semantic-release` is effectively broken on CircleCI right now, and there's no telling if/when the `npm` fix will ship.

**Comment by bahmutov** [comment]
@motiz88 will do, opened https://github.com/bahmutov/condition-circle/issues/3

**Comment by krishna-koushik** [comment]
Not sure if i did this right but it did not work for me.

<img width="1315" alt="screen shot 2016-08-15 at 2 35 52 pm" src="https://cloud.githubusercontent.com/assets/622622/17674965/bb739280-62f5-11e6-92cc-699e74240c75.png">

My Circle.yml looks like this at the end.

<img width="1004" alt="screen shot 2016-08-15 at 2 37 27 pm" src="https://cloud.githubusercontent.com/assets/622622/17674993/d3ccb32a-62f5-11e6-88e2-7c93bf13e3a8.png">

**Comment by bahmutov** [comment]
@motiz88 I have updated condition-circle to run the script as part of the condition, seems to work - now my modules on CircleCI are actually bumped correctly.

**Comment by motiz88** [comment]
@krishna-koushik I think you did it right. The fix takes effect from the *next* publish. To verify this without pushing another version right now, you can look at the output of `npm info @meltwater/output-manager-api` and see that it has a non-empty `gitHash` field.

@bahmutov Fantastic :smile:

**Comment by tleunen** [comment]
Glad you found a way to fix it @motiz88!

**Comment by krishna-koushik** [comment]
I ran the command but i did not see a gitHash field.

`gitHead: 'b355667a2a2a83b8ae7dd1c4f30fbb03d9c66643',
  dist: 
   { shasum: '7959dd9490983e39777e9873932611d394d9b234',
     tarball: '' },
  directories: {} }`

**Comment by motiz88** [comment]
@krishna-koushik I meant `gitHead`, so you're fine. (Edited my previous comment accordingly)

**Comment by krishna-koushik** [comment]
Awesome thank you @motiz88 :)

**Comment by peteyy** [comment]
I added the following to the circle.yaml:

```yaml
deployment:
  master:
    branch: master
    commands:
      - git show-ref --head --heads | while IFS=' ' read -r hash name; do test ! -e "${GIT_DIR:-.git}/$name" && echo $hash > "${GIT_DIR:-.git}/$name"; done
      - npm run semantic-release
```

But I still got a major release out of a `feat()` tag without a `BREAKING` line.
